analysis/syntactic/structinit/structinit.go

```
@@ -822,3 +853,139 @@ func isFiltered(spec config.StructInitSpec, f *ssa.Function) bool {
 
 	return false
 }
+
+// runMustReinitChecks runs all the must-reinit checks and returns a map from named struct type to
+// a possibly empty list of problems of bad reinitialization of that struct.
+func runMustReinitChecks(
+	state *ptr.State,
+	specs []config.StructInitSpec,
+	fns map[*ssa.Function]bool,
+	structToNamed map[*types.Struct]*types.Named) map[*types.Named][]BadReinit {
+	state.Logger.Infof("Run must-reinit checks.")
+	badReinits := map[*types.Named][]BadReinit{}
+	for fn := range fns {
+		if summaries.IsStdPackageName(lang.PackageNameFromFunction(fn)) {
+			continue
+		}
+		lang.IterateInstructions(fn, func(_ int, instr ssa.Instruction) {
+			if instr == nil || instr.Parent() == nil || !instr.Pos().IsValid() {
+				return
+			}
+
+			if call, isCall := instr.(*ssa.Call); isCall {
+				if !lang.CanType(call) {
+					return
+				}
+				namedType := checkStructTyp(call.Type(), structToNamed)
+				if namedType == nil {
+					return
+				}
+				if maybeBadReinit := checkMustReinitCall(state, specs, call); maybeBadReinit.IsSome() {
+					if _, ok := badReinits[namedType]; !ok {
+						badReinits[namedType] = []BadReinit{}
+					}
+					badReinits[namedType] = append(badReinits[namedType],
+						maybeBadReinit.Value())
+				}
+			}
+		})
+	}
+	return badReinits
```

Should there be an error if there is a `must-reinit` spec but `len(badReinits) == 0`? The analysis would silently succeed in the trivial case where `len(fns) == 0`

---

This is also true for all the analyses, they succeed when there is nothing to analyze and I it should  not return an error. Some analyses are run as "sanity checks" just in case someone introduced something that needs analyzing. 
Sanity checking that the analyses are running should be a different mechanism (e.g. introduce changes that should trigger alarms).